### PR TITLE
Implement object transformers

### PR DESCRIPTION
**Pull request by cthulhu-rider**
## Summary

- define interface of the object writer;

- implement payload size limiter;

- implement object structure moulder;

- accompanying `core/object` changes.

### CODE REVIEW

File: pkg/services/object_manager/transformer/types.go [hunk 1]
```diff
@@ -0,0 +1,90 @@
...
+	// Must be called exactly once. Control remains with the caller.
+	// Missing a call or re-calling can lead to undefined behavior
+	// that depends on the implementation.
+	//
+	// Must not be called after Close call.
+	WriteHeader(*object.RawObject) error
+
+	// Write writes object payload chunk.
+	//
+	// Can be called multiple times. Must not be called
```
Comment (alexvanin) [inline pkg/services/object_manager/transformer/types.go:31]: Remove `Must not be called`, explained in next line.

File: pkg/services/object_manager/transformer/fmt.go [hunk 2]
```diff
@@ -0,0 +1,97 @@
...
+	"crypto/ecdsa"
+
+	"github.com/nspcc-dev/neofs-api-go/pkg"
+	objectSDK "github.com/nspcc-dev/neofs-api-go/pkg/object"
+	"github.com/nspcc-dev/neofs-node/pkg/core/object"
+	"github.com/pkg/errors"
+)
+
+type formatter struct {
+	nextTarget ObjectTarget
```
Comment (alexvanin) [inline pkg/services/object_manager/transformer/fmt.go:13]: Pipeline with the moulder will look like `moulder` -> `formatter` -> `sizeLimiter` -> `formatter`?

**Review by alexvanin — APPROVED** [review]
Look forward for use cases at object service.

File: pkg/local_object_storage/localstore/meta.go [hunk 3]
```diff
@@ -63,12 +61,12 @@ func metaToBytes(m *ObjectMeta) ([]byte, error) {
 
 	binary.BigEndian.PutUint64(data, m.savedAtEpoch)
```
[the remark this answers is not in the document]
Comment (cthulhu-rider) [reply pkg/local_object_storage/localstore/meta.go:62]: No, we have creation epoch number in `m.head`, and `m.savedAtEpoch` is the epoch number in which object was stored locally.

File: pkg/services/object_manager/transformer/fmt.go (hunk 2 above)
Comment (cthulhu-rider) [reply pkg/services/object_manager/transformer/fmt.go:13]: Just `transformer` -> `moulder`.

File: pkg/services/object_manager/transformer/types.go (hunk 1 above)
Comment (cthulhu-rider) [reply pkg/services/object_manager/transformer/types.go]: Fixed

**Pull request merged**